utils.bal

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+import ballerina/http;
+import ballerina/time;
+
+isolated function buildPayload(map<string> parameters) returns string {
+    string payload = EMPTY_STRING;
+    int parameterNumber = 1;
+    foreach var [key, value] in parameters.entries() {
+        if (parameterNumber > 1) {
+            payload = payload + AMBERSAND;
+        }
+        payload = payload + key + EQUAL + value;
+        parameterNumber = parameterNumber + 1;
+    }
+    return payload;
+}
+
+isolated function buildQueryString(string actionName, map<string> parameterMap, string... parameterValues) returns @tainted map<string> {
+    int index = 0;
+    string parameterValueString = "";
+    parameterMap[ACTION] = actionName;
+    parameterMap[VERSION] = VERSION_NUMBER;
+    foreach string? parameterValue in parameterValues {
+        parameterMap[getAttributeName(parameterValue)] = parameterValue;            
+        index += 1;
+    }
+    return parameterMap;
+}
+
+isolated function addOptionalStringParameters(map<string> parameterMap, string?... optionalParameterValues) returns @tainted map<string>|error {
+    int index = 0;
+    string parameterValueString = "";
+    foreach string? optionalParameterValue in optionalParameterValues {
+        if(optionalParameterValue is string) {
+            parameterMap[getAttributeName(optionalParameterValue)] = optionalParameterValue;            
+        }
+        index += 1;
+    }
+    return parameterMap;
+}
+
+isolated function sendRequest(http:Client amazonSimpleDBClient, http:Request|error request, string queryParameters) returns @tainted xml|error {
+    if (request is http:Request) {
+        http:Response|error httpResponse = amazonSimpleDBClient->post("/", request);
+        xml|error response = handleResponse(httpResponse);
+        if (response is xml) {
+            return response;
+        } else {
+            return response;
+        }
+    } else {
+        return error(REQUEST_ERROR);
+    }
+}
+
+isolated function validateCredentials(string accessKeyId, string secretAccessKey) returns error? {
+    if ((accessKeyId == "") || (secretAccessKey == "")) {
+        return error("Access Key Id or Secret Access Key credential are empty");
```

Comment:
We could make these strings as constants in a separate file then it will be easy to maintain. isn't it?

Reply:
Added to constant